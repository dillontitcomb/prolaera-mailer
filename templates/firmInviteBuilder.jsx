import React from 'react';
import { Email, renderEmail } from 'react-html-email';
import FirmInvite from '../components/firmInvite/firmInvite';
import InviteSubHeader from '../components/firmInvite/inviteSubHeader';
import Footer from '../components/footer';
import builderHeader from '../components/header';
import css from './templateCSS.js';

const firmInviteBuilder = async (imageUrl, completeUser, completeCompany) => {
  try {
    const Header = await builderHeader(imageUrl);
    return renderEmail(
      <Email title="" headCSS={css}>
        <Header />
        <InviteSubHeader />
        <FirmInvite {...completeUser} {...completeCompany} />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};

export default firmInviteBuilder;
