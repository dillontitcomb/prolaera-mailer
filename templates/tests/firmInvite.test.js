import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import fs from 'fs';
import React from 'react';
import renderer from 'react-test-renderer';
import FirmInvite from '../../components/firmInvite/firmInvite';
import firmInviteBuilder from '../firmInviteBuilder';

Enzyme.configure({ adapter: new Adapter() });
const user = { email: 'test@email.com' };
const company = { company_name: 'Test CPA Firm' };

describe('create html test file', () => {
  it('it writes an html file to test.html', async () => {
    const firmInviteEmail = await firmInviteBuilder(
      'https://assets.prolaera.com/prolaeraLogo_fullText.png',
      user,
      company
    );
    const saved = await writeFile(firmInviteEmail);
    expect(saved).toEqual(true);
  });
});

describe('Firm Invite component', () => {
  it('creates and checks snapshot of FirmInvite component html', async () => {
    let inviteHtml = renderer.create(<FirmInvite {...user} {...company} />);
    let inviteJson = inviteHtml.toJSON();
    expect(inviteJson).toMatchSnapshot();
  });

  it('checks FirmInvite custom email and company_name inputs', async () => {
    const wrapper = shallow(<FirmInvite {...user} {...company} />);
    expect(
      wrapper.contains(
        <a
          className="accountButton"
          href="https://app.prolaera.com/#/registration?email=/test@email.com"
          style={{
            backgroundColor: '#72C02C',
            color: 'white',
            maxWidth: '200px',
            padding: '14px 14px 14px 14px',
            textDecoration: 'none'
          }}
        >
          Create My Account
        </a>
      )
    ).toBe(true);
    expect(wrapper.contains(<p>You have been added to Test CPA Firm's team on Prolaera!</p>)).toBe(true);
  });
});

async function writeFile(emailHtml) {
  return new Promise((resolve, reject) => {
    fs.writeFile(`${__dirname}/firmInviteTest.html`, emailHtml, err => {
      if (err) {
        console.log(err);
        reject(err);
      }
      resolve(true);
    });
  });
}
