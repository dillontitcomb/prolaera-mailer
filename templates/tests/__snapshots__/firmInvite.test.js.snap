// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Firm Invite component creates and checks snapshot of FirmInvite component html 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "display": "block",
      "fontFamily": "Arial, sans-serif",
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": "20px",
      "textAlign": "center",
      "width": "500px",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    className="firmInvitation"
    valign="top"
    width="500px"
  >
    <tbody>
      <tr>
        <td
          align="left"
        >
          <p>
            You have been added to 
            Test CPA Firm
            's team on Prolaera!
          </p>
        </td>
      </tr>
      <tr>
        <td
          align="left"
        >
          <p>
            Your account is waiting for you and there are only a couple quick steps to complete.
          </p>
          <ol>
            <li>
              Click the button below to register your account
            </li>
            <li>
              On the registration page simply enter a password for your new account and click "Register".
            </li>
          </ol>
          <p>
            That's it! You're ready to explore the new tools and resources you have access to.
          </p>
          <div
            className="buttonContainer"
            style={
              Object {
                "padding": "20px 0px 20px 0px",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <a
              className="accountButton"
              href="https://app.prolaera.com/#/registration?email=/test@email.com"
              style={
                Object {
                  "backgroundColor": "#72C02C",
                  "color": "white",
                  "maxWidth": "200px",
                  "padding": "14px 14px 14px 14px",
                  "textDecoration": "none",
                }
              }
            >
              Create My Account
            </a>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
